packages/headless/src/controllers/search-box/headless-search-box.ts

```diff
@@ -156,8 +156,7 @@ export function buildSearchBox(
   };
 
   validateOptions(engine, searchBoxOptionsSchema, options, 'buildSearchBox');
-
-  dispatch(registerQuerySetQuery({id, query: ''}));
+  dispatch(registerQuerySetQuery({id, query: engine.state.query.q}));
```

Comment:
was empty here, but filled with `engine.state.query.q` for "registerQuerySuggest" ?

Reply:
Why should the new search box be initialized to the query? I'd expect it to start off empty, synchronizing when a executeSearch is fulfilled,

Reply:
If an execute search is executed before the component is registered, the newly created searchBox should represent this state?
Also as mentioned `registerQuerySuggest` did this already